### PR TITLE
Parametrize datasources

### PR DESCRIPTION
**Pull request by wcandillon**

### CODE REVIEW

**Comment by wcandillon** [comment]
@gislenius Does it achieve what you want?
@dknochen Is it clean?

File: tasks/templates.js [hunk 1]
```diff
@@ -28,6 +29,14 @@ gulp.task('templates:create', ['config:load'], function(done){
         elasticSearchMapping = fs.readFileSync(mappingsFile, 'utf-8');
     }
 
+    //Indicates which datasource is for query and which one is for search
+    var queryDatasource  = _.chain(Config.credentials['28'].datasources)
+            .filter(function(source){ return source.type === 'query'; })
```
Comment (dknochen) [inline tasks/templates.js]: if we make the type field an array, we could use the same datasource for query and search.

File: tasks/templates.js [hunk 2]
```diff
@@ -28,6 +29,14 @@ gulp.task('templates:create', ['config:load'], function(done){
...
+            .map(function(source){ return _.template('{ name: "<%= name %>", category: "<%= category %>" }')(source); }).value() || '()';
+    var searchDatasource = _.chain(Config.credentials['28'].datasources)
+            .filter(function(source){ return source.type === 'search'; })
+            .map(function(source){ return _.template('{ name: "<%= name %>", category: "<%= category %>" }')(source); }).value() || '()';
```
Comment (dknochen) [inline tasks/templates.js]: this will never be '()' because value() returns an empty array. We need to check for an empty array explicitly

Comment (dknochen) [inline tasks/templates.js]: we should do  ....value()[0] || '()'

File: tasks/templates/config.jq.mustache [hunk 3]
```diff
@@ -27,4 +27,8 @@ declare variable $config:sendmail-sender-name as string := "{{sendmail.sender.na
 declare variable $config:profile-name as string := "{{cellstore.profile}}";
 declare variable $config:filtered-aspects as integer := {{cellstore.filteredAspects}};
 
+declare variable $config:query-datasource := {{& queryDatasource }};
```
Comment (gislenius) [inline tasks/templates/config.jq.mustache]: can you add a type here?

Comment (gislenius) [inline tasks/templates/config.jq.mustache]: Technically, in the config file, I would need both the category and name of the source for both query and search. So are these objects with two fields?

File: tasks/templates.js (hunk 1 above)
Comment (wcandillon) [inline tasks/templates.js]: @dknochen can you elaborate?

File: tasks/templates.js (hunk 2 above)
Comment (wcandillon) [inline tasks/templates.js]: got it

**Comment by wcandillon** [comment]
@dknochen Implemented the review comments

**Comment by dknochen** [comment]
looks good :+1:

**Comment by dknochen** [comment]
@gislenius approved from my side.

**Comment by gislenius** [comment]
@dknochen from mine as well. Do you merge?

**Pull request merged**